internal/mock/cloudsql.go

```diff
@@ -30,45 +35,43 @@ type FakeCSQLInstance struct {
 	project string
 	region  string
 	name    string
-
-	dbVersion string
-
-	privKey *rsa.PrivateKey
+	version string
+	key     *rsa.PrivateKey
 	cert    *x509.Certificate
+	tlsCert tls.Certificate
 }
 
 // NewFakeCSQLInstance returns a CloudSQLInst object for configuring mocks.
-func NewFakeCSQLInstance(project, region, name string) (FakeCSQLInstance, error) {
+func NewFakeCSQLInstance(project, region, name string) FakeCSQLInstance {
 	// TODO: consider options for this?
-	privKey, cert, err := generateInstanceCerts()
+	key, cert, tlsCert, err := generateCerts(project, name)
 	if err != nil {
-		return FakeCSQLInstance{}, err
+		panic(err)
 	}
 
-	c := FakeCSQLInstance{
-		project:   project,
-		region:    region,
-		name:      name,
-		dbVersion: "POSTGRES_12", // default of no particular importance
-		privKey:   privKey,
-		cert:      cert,
+	return FakeCSQLInstance{
+		project: project,
+		region:  region,
+		name:    name,
+		version: "POSTGRES_12", // default of no particular importance
+		key:     key,
+		cert:    cert,
+		tlsCert: tlsCert,
 	}
-	return c, nil
 }
 
-// generateInstanceCerts returns a key and cert for representing a Cloud SQL instance.
-func generateInstanceCerts() (*rsa.PrivateKey, *x509.Certificate, error) {
-	privKey, err := rsa.GenerateKey(rand.Reader, 4096)
+// generateCerts generates a private key, an X.509 certificate, and a TLS
+// certificate for a particular fake Cloud SQL database instance.
+func generateCerts(project, name string) (*rsa.PrivateKey, *x509.Certificate, tls.Certificate, error) {
+	key, err := rsa.GenerateKey(rand.Reader, 2048)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, tls.Certificate{}, err
 	}
 
 	cert := &x509.Certificate{
 		SerialNumber: &big.Int{},
 		Subject: pkix.Name{
-			Country:      []string{"US"},
-			Organization: []string{"Google, Inc"},
-			CommonName:   "Google Cloud SQL Server CA",
+			CommonName: fmt.Sprintf("%s:%s", project, name),
```

Comment:
Is this right? I thought the region would be included in the common name.

Reply:
It wouldn't surprise me if it's left off considering region isn't used to identify the instances

Reply:
I confirmed with openssl against my own database and see that it's correct.